File: urbanairship-core/src/main/java/com/urbanairship/Predicate.java
/* Copyright Airship and Contributors */

package com.urbanairship;

import android.annotation.SuppressLint;

/**
 * Predicate interface.
 *
 * @param <T> predicate type.
 */
public interface Predicate<T> {

    /**
     * Applies the predicate against an object.
     *
     * @param object The object.
     * @return {@code true} if the predicate matches the object, otherwise {@code false}.
     */
    @SuppressLint("UnknownNullness")
    boolean apply(T object);

}

File: urbanairship-core/src/main/java/com/urbanairship/Predicate.kt
/* Copyright Airship and Contributors */
package com.urbanairship

/**
 * Predicate interface.
 *
 * @param T The type of the predicate value.
 */
public fun interface Predicate<T: Any> {

    /**
     * Applies the predicate against a value.
     *
     * @param value The value to test this predicate against.
     * @return `true` if the predicate matches the object, otherwise `false`.
     */
    public fun apply(value: T): Boolean
}
